apis/cassandra/v1beta1/cassandradatacenter_webhook.go

```diff
@@ -243,3 +253,51 @@ func ValidateFQLConfig(dc CassandraDatacenter) error {
 
 	return nil
 }
+
+func ValidateServiceLabelsAndAnnotations(dc CassandraDatacenter) error {
+	// check each service
+	addSeedSvc := dc.Spec.AdditionalServiceConfig.AdditionalSeedService
+	allPodsSvc := dc.Spec.AdditionalServiceConfig.AllPodsService
+	dcSvc := dc.Spec.AdditionalServiceConfig.DatacenterService
+	nodePortSvc := dc.Spec.AdditionalServiceConfig.NodePortService
+	seedSvc := dc.Spec.AdditionalServiceConfig.SeedService
+
+	services := map[string]ServiceConfigAdditions{
+		"AdditionalSeedService": addSeedSvc,
+		"AllPOdsService":        allPodsSvc,
+		"DatacenterService":     dcSvc,
+		"NodePOrtService":       nodePortSvc,
+		"SeedService":           seedSvc,
+	}
+
+	var msg string
+
+	for svcName, config := range services {
+		if containsReservedAnnotations(config) || containsReservedLabels(config) {
+			msg = string(append([]byte(msg), fmt.Sprintf("configure %s with reserved annotations and/or labels (prefixes %s and/or %s)", svcName, datastaxPrefix, k8ssandraPrefix)...))
+		}
+	}
+	msg = strings.Trim(msg, " ")
```

Comment:
This is probably due to the odd casting. I think len(msg) > 0 should be enough. Or just bail out after first error.